script/jenkins/build.sh

```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
 set -xeuo pipefail
 
-mage -debug build
+make apm-server
```

Comment:
we preferred to use mage build it is simple and does not have any no-build stuff

Reply:
What does "no-build stuff" mean?

Reply:
BTW, the previous Makefile was hideously complex, so I can understand why you wouldn't have wanted to rely on it. Take a look at the new one, it's a lot simpler now.

Reply:
I am not sure I understand your concern, the `make` targets to which I changed to the same as the formerly used `mage` targets.

Reply:
the concern is about that we move away from `make` to avoid make stuff not related to build/compile/test, the previous Makefile made some checks about tools install those tools and finally starts to compile/build/test, we want to avoid this and manage the things isolated.

Reply:
OK. The Makefile does setup tools, but it does so in a repeatable and deterministic manner, based on on the versions specified in `go.mod`. So the end result is the same, whether you install the tools first and then run `mage` directly, or run `make` which installs the tools and then runs `mage`.

I'd rather have one single build procedure that both CI and devs use. Please take a look at the new Makefile and let us know if you still have concerns.

Reply:
I saw it, we download the modules, then we run a  `make` target to build `mage`, and then we can use `mage`, complex form my POV

Reply:
To be fair, the `go mod download` isn't necessary; it's an optimisation to avoid re-downloading each time the container is run. We could remove that and the rest would work, just more slowly.

Apart from that, the idea is to have everything encapsulated within the Make command: installing and running the required tools, in order to build the targets. This takes care of bootstrapping the build environment, to minimise the number of tools required in the environment to begin with. You need Go, Python, Make (`build-essential`), and that's about it.

(From the other comment thread)

> The new Makefile is much simpler. What makes me think about why we do not move everything to mage and remove the Makefile? this makes mage a pre-requisite but wi will have only one build definition magefile.go and only one build tool mage

A couple of reasons:
 - Make is a standard (and very mature) development tool in Linux, Mage is not.
 - If we require Mage to be installed globally, it makes it more difficult to rely on a specific version. It's true that we don't rely on a specific version of _Make_, but again, that's vastly more mature.

We can of course control the version in CI, but that doesn't help on developers' laptops.

Anyway, I do not see what the issue is here since we would be maintaining the Makefile either way, but if it's really a problem for you/CI then we can globally install and directly call mage just in CI. My concern is that the more you diverge CI from day-to-day development, the more likely it is that issues will escape dev and show up in CI.

Reply:
>If we require Mage to be installed globally, it makes it more difficult to rely on a specific version. It's true that we don't rely on a specific version of Make, but again, that's vastly more mature.

You require it there are make targets that does not work without mage

>Anyway, I do not see what the issue is here since we would be maintaining the Makefile either way, but if it's really a problem for you/CI then we can globally install and directly call mage just in CI. 

you can make whatever you want with the makefile, mage file, python, ... 

>My concern is that the more you diverge CI from day-to-day development, the more likely it is that issues will escape dev and show up in CI.

Give us the target we have to execute from the CI to build and test the project, we will manage to execute them based on our need for the CI.